intake/readers/readers.py

```diff
@@ -1018,7 +1018,11 @@ def _read(self, data, **kw):
             return open_mfdataset(data.url, **kw)
         else:
             # TODO: recognise fsspec URLs, and optionally use open_local for engines tha need it
-            if isinstance(data, datatypes.FileData) and data.url.startswith("http"):
+            if (
+                isinstance(data, datatypes.FileData)
+                and data.url.startswith("http")
+                and not datatypes.Zarr
```

Comment:
I think:
```suggestion
                and not isinstance(data, datatypes.Zarr)
```

`not datatypes.Zarr` is always `False`, so you would never hit the DAP endpoint.

Reply:
Sorry about that, I should have stopped to make a test. I am working on that now.

Reply:
Well, it is hard to make a test without sample data.

Reply:
I made your change and have the start of a test but don't have any good zarr vs opendap example data to use to differentiate between the two http cases to use. Do you have any suggestions?

Reply:
A local server perhaps? You can then write some data to a temporary directory.

Reply:
My sneaky attempt to move more folks away from OpenDAP has been foiled.